### DISCUSSION TITLE
Export to asciinema video

### DISCUSSION THREAD

**Discussion by Jonathan-Zollinger** [charmbracelet/vhs#225] [Ideas]
Itd be nice imo to be able to export VHS tape as asciinema videos since these videos can be played in the terminal. I feel it has some great potential for tui integrations, embracing asciinema compatibility

**Comment by taigrr** [comment]
Seconded, but note this is a heavier lift, as we can't just include an asciinema library, asciinema is a python application. Would require writing a new version of asciinema in go to pull from.

**Reply by Jonathan-Zollinger** [reply]
woof, that's a big undertaking.

**Reply by Jonathan-Zollinger** [reply]
I see there's a really old golang branch in asciinema's repo. it's not compatible with the current golang version - I'm new to Go, but what better way to learn than throwing yourself head first into foreign code that doesn't work?

**Comment by bashbunni** [comment]
This is a really interesting idea. I agree it would be great to have an output option that can be viewed in the terminal :thought_balloon:

**Comment by garbas** [comment]
I don't think it would be that hard to provide asciinema outpout since the format specification is pretty straightforward -> https://github.com/asciinema/asciinema/blob/develop/doc/asciicast-v2.md

I'd do it myself, but I'm not so familiar (yet) with the vhs codebase.

**Comment by drupol** [comment]
Hello, is there any progress on this feature?

**Reply by taigrr** [reply]
You can check my feature branch if you want to try it out

asciicinema works very differently from vhs though, so my implementation matches vhs, i.e. creating a fixed canvas. Asciicinema allows for resizes mid-record etc.

**Reply by drupol** [reply]
Link?

**Reply by drupol** [reply]
@taigrr got a link please?

**Reply by taigrr** [reply]
it's on my fork, here

https://github.com/taigrr/vhs/tree/clawd/asciinema-output

**Reply by drupol** [reply]
Hello,

I have used your patch and added it in a Nix Hands On tutorial.

https://github.com/drupol/nix-hands-on Check step 17.

I will now make some tests with it.

Question: have you already tried to submit this patch upstream ?

**Reply by drupol** [reply]
Here's the result:

- Asciinema output: https://asciinema.org/a/XKdxhbm5Y4cgZ27Z
- Gif output: ![demo](https://github.com/user-attachments/assets/cad9b7f2-5094-4624-a85b-26799fc6c22e)

Script:

```
Output demo.cast
Output demo.gif

Set Shell fish

Type "nix run nixpkgs#cowsay -- vhs with asciinema output is cool !"

Sleep 500ms

Enter

Sleep 2s
```

**Reply by taigrr** [reply]
no i havent. @meowgorithm is that something you'd be interested in having upstreamed?

**Reply by meowgorithm** [reply]
Yes!

**Reply by taigrr** [reply]
pr #706
